### PR TITLE
TypeScript Cleanup - Bye, bye TODOs 

### PR DESCRIPTION
**Pull request by nickytonline**
@bfred-it, just some more post #1735 TODO cleanup. I removed the tiny-version-compare types. The package needs to be bumped in the project, but just waiting for https://github.com/bfred-it/tiny-version-compare/pull/7 to be merged before going ahead.

### CODE REVIEW

**Comment by bfred-it** [comment]
Instead of many tiny PRs let’s reuse this one for any TS-relate changes this week

File: source/globals.d.ts [hunk 1]
```diff
@@ -1,12 +1,5 @@
 type AnyObject = Record<string, any>;
 
-// TODO: Move types to tiny-version-compare repo
-declare module 'tiny-version-compare' {
-	function compareVersions(versionA: string, versionB: string): number
-
-	export = compareVersions
-}
-
 // TODO: Move types to https://github.com/sindresorhus/linkify-urls repository.
```
Comment (bfred-it) [inline source/globals.d.ts]: This TODO should be updated to point to our last discussion about v3 compatibility

**Comment by nickytonline** [comment]
This should be good to go. TODOs that remain have had comments updated.

**Comment by bfred-it** [comment]
Any other TODOs across the extension? There are quite a few type-related comments I think

**Comment by nickytonline** [comment]
Ahh, yeah. True that. I had only globals.d.ts on the brain...

**Comment by bfred-it** [comment]
I'm surprised to see that the todos are actually already gone. The last one would be this:


```ts
// TODO: type target asap and drop `as Element`
const observer = new MutationObserver(([{target}]) => {
	if (!select.exists('.js-details-dialog-spinner', target as Element)) {
		improveShortcutHelp(target as Element);
		fixKeys(target as Element);
		observer.disconnect();
	}
});
```

Is it possible without adding `const dialog: Element = target`?

**Comment by nickytonline** [comment]
I was working on the last todo last  night. No success yet but basically the `MutationCallback` interface or`MutationRecord`  interface needs to be altered via a merge declaration. Altering a `MutationRecord` to make `target` an `Element` doesn't seem to work and I was having issues changing the `MutationCallback`.

I'll take another stab at it tonight and report back.

**Comment by bfred-it** [comment]
Are you trying to change the type of `target` globally? I don’t think that’s right because `target` can also be a regular text node. 

I think a shortcut would be to add `&& target instanceof Element` to the existing `if`

**Comment by nickytonline** [comment]
Alright, a little lunchtime break/commit and I think this is good to go @bfred-it.

**Comment by bfred-it** [comment]
Excellent!

**Comment by bfred-it** [comment]
I guess the last step would be to also convert the `webpack.config.js` (which I was initially against) but it's not necessary. I just pushed abca820 to avoid some TS/eslint conflicts in there

**Comment by nickytonline** [comment]
Happy to convert it if you want. I've done it a bunch of times before.

**Comment by bfred-it** [comment]
# ✌️

**Pull request merged**